pex/environment.py

```diff
@@ -49,7 +50,11 @@ def force_local(cls, pex, pex_info):
           safe_rmtree(explode_tmp)
           raise
       TRACER.log('Renaming %s to %s' % (explode_tmp, explode_dir))
-      os.rename(explode_tmp, explode_dir)
+      try:
```

Comment:
How about using this which is slightly more correct (it cleans up on rename failure) and already tested - so you can maybe drop your test:
https://github.com/pantsbuild/pex/blob/47c26d746046a8ab21bd3cb3bb782ce5f018a369/pex/common.py#L160-L172